src/Uno.Toolkit.UI/Controls/NavigationBar/NavigationBarRenderer.Android.cs

```diff
@@ -164,6 +160,12 @@ protected override IEnumerable<IDisposable> Initialize()
 					new[] { NavigationBar.MainCommandProperty, AppBarButton.IconProperty },
 					new[] { NavigationBar.MainCommandModeProperty }
 				);
+
+
+				yield return (element.MainCommand?.Icon as BitmapIcon).SubscribeNestedPropertyChangedCallback(OnPropertyChanged);
```

Comment:
this is not how you would do nested property subscription; given that initialize is only run once/or at every render(?), if the these brushes has since changed and then their nested properties changes wont be listened by _anyone_

Reply:
Hmm yeah you're right. Perhaps just another DependencyObject extension like `RegisterNestedPropertyChange<T>(this DependencyObject object, string[][] properties)`?

Where it would take something like AppBarButton.Icon and attempt to cast to T, which would be BitmapIcon in this case?